src/SW_netCDF.c

```diff
@@ -0,0 +1,1941 @@
+                              SW_netCDF->primary_crs_is_geographic,
+                              SW_Domain->DomainType, LogInfo);
+    if(LogInfo->stopRun) {
+        nc_close(*domFileID);
+        return; // Exit function prematurely due to error
+    }
+
+    fill_netCDF_with_invariants(SW_netCDF, SW_Domain->DomainType,
+                                domFileID, swTRUE, LogInfo);
+    if(LogInfo->stopRun) {
+        nc_close(*domFileID);
+        return; // Exit function prematurely due to error
+    }
+
+    nc_enddef(*domFileID);
+
+    fill_domain_netCDF_vals(SW_Domain, *domFileID, domVarID, sVarID,
+                            latVarID, lonVarID, latBndsID, lonBndsID,
+                            yBndsID, xBndsID, LogInfo);
+
+    nc_close(*domFileID);
+}
+
+/**
+ * @brief Copy domain netCDF to a new file and make it ready for a new variable
+ *
+ * @param[in] fileName Name of the netCDF file to create
+ * @param[in] timeSize Size of "time" dimension
+ * @param[in] vertSize Size of "vertical" dimension
+ * @param[in] varName Name of variable to write
+ * @param[in] varAttributes Attributes that the new variable will contain
+ * @param[in,out] LogInfo  Holds information on warnings and errors
+*/
+void SW_NC_create_template(const char* fileName, unsigned long timeSize,
+                           unsigned long vertSize, const char* varName,
+                           char* varAttributes[], LOG_INFO* LogInfo) {
+    (void) fileName;
+    (void) timeSize;
+    (void) vertSize;
+    (void) varName;
+    (void) varAttributes;
+    (void) LogInfo;
+}
+
+/**
+ * @brief Read values from netCDF input files for available variables and copy to SW_All
+ *
+ * @param[in,out] sw Comprehensive struct of type SW_ALL containing
+ *  all information in the simulation
+ * @param[in] SW_Domain Struct of type SW_DOMAIN holding constant
+ *  temporal/spatial information for a set of simulation runs
+ * @param[in] ncSUID Current simulation unit identifier for which is used
+ *  to get data from netCDF
+ * @param[out] LogInfo Holds information on warnings and errors
+*/
+void SW_NC_read_inputs(SW_ALL* sw, SW_DOMAIN* SW_Domain, size_t ncSUID[],
+                       LOG_INFO* LogInfo) {
+
+    int file, varNum;
+    char* geo_long_name = SW_Domain->netCDFInfo.crs_geogsc.long_name;
+    Bool geoFilled = Str_CompareI(SW_Domain->crs_bbox, geo_long_name) == 0 ||
+                     (is_wgs84(SW_Domain->crs_bbox) && is_wgs84(geo_long_name));
+    Bool domTypeS = Str_CompareI(SW_Domain->DomainType, "s") == 0;
+    const int numDomVals = 2;
+    const int numVals[] = {numDomVals};
+    const int ncFileIDs[] = {SW_Domain->netCDFInfo.ncFileIDs[DOMAIN_NC]};
+    const char* domLatVar = (geoFilled) ? "lat" : "y";
+    const char* domLonVar = (geoFilled) ? "lon" : "x";
+    const char* varNames[][2] = {{domLatVar, domLonVar}};
+    int ncIndex;
+
+    RealD *values[][2] = {{&sw->Model.latitude, &sw->Model.longitude}};
```

Comment:
in `SW_NC_read_inputs()`: 
the values of &sw->Model.latitude and &sw->Model.longitude are always obtained from (geographic horizontal variables) "lat" and "lon" and never from (projected horizontal variables) "x" and "y" (because "x" and "y" are in unsuitable CRS); that is why we enforce the presence of a geographic CRS